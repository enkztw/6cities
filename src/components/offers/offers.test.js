import React from 'react';
import renderer from 'react-test-renderer';
import Offers from './offers.jsx';

const city = `Samara`;
const cities = [`Samara`, `Moscow`, `Palo Alto`];

const offers = [
  {
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    images: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    bedrooms: 3,
    adults: 4,
    price: 110,
    premium: false,
    rating: 4.51,
    bookmarked: true
  },
  {
    name: `Wood and stone place`,
    type: `Room`,
    images: [`img/apartment-01.jpg`, `img/room.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    bedrooms: 1,
    adults: 2,
    price: 120,
    premium: true,
    rating: 4.49,
    bookmarked: true
  },
  {
    name: `Canal View Prinsengracht`,
    type: `House`,
    images: [`img/apartment-02.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    bedrooms: 3,
    adults: 3,
    price: 240,
    premium: false,
    rating: 2.11,
    bookmarked: false
  },
  {
    name: `Nice, cozy, warm big bed apartment`,
    type: `Hotel`,
    images: [`img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    bedrooms: 1,
    adults: 2,
    price: 50,
    premium: true,
    rating: 3.2,
    bookmarked: false
  }];

describe(`Render <Main />`, () => {
  it(`<Offers /> should render offer's container`, () => {
    const tree = renderer.create(<Offers cities={cities} offers={offers} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`<Offers /> should render empty container`, () => {
    const tree = renderer.create(<Offers cities={cities} offers={[]} city={city} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
