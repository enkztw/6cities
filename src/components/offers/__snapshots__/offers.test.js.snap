// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Main /> <Offers /> should render empty container 1`] = `
<main
  className="page__main page__main--index page__main--index-empty page__main--index-empty"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__itemtabs__item tabs__item--active"
            href="#"
          >
            <span>
              Samara
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
          >
            <span>
              Moscow
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
          >
            <span>
              Palo Alto
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container cities__places-container--empty container"
    >
      <section
        className="cities__no-places"
      >
        <div
          className="cities__status-wrapper tabs__content"
        >
          <b
            className="cities__status"
          >
            No places to stay available
          </b>
          <p
            className="cities__status-description"
          >
            We could not find any property availbale at the moment in 
            Samara
          </p>
        </div>
      </section>
      <div
        className="cities__right-section"
      />
    </div>
  </div>
</main>
`;
